lib/sendgrid/base_interface.rb

```diff
@@ -16,9 +16,15 @@ class BaseInterface
   #                              in the "On-Behalf-Of" header
   #   - +http_options+ -> http options that you want to be globally applied to each request
   #
-  def initialize(auth:, host:, request_headers: nil, version: nil, impersonate_subuser: nil, http_options: {})
+  def initialize(auth:, host:, region: nil, request_headers: nil, version: nil, impersonate_subuser: nil, http_options: {})
     @auth = auth
-    @host = host
+    if host.nil? && region.nil?
```

Comment:
If host is not default to `nil`, can it be `nil` now? 
How about this? 

```
if !region.nil?
      data_residency(region: region)
    else
      @host = host
```

Reply:
adressed